src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs

```diff
@@ -119,11 +119,12 @@ public async Task ConnectWithCertificateChain()
         public async Task WaitForAvailableUnidirectionStreamsAsyncWorks()
         {
             using QuicListener listener = CreateQuicListener(maxUnidirectionalStreams: 1);
-            using QuicConnection clientConnection = CreateQuicConnection(listener.ListenEndPoint);
+            QuicConnection clientConnection = CreateQuicConnection(listener.ListenEndPoint);
```

Comment:
Why did you swap the `using` for explicit `Dispose`?

Reply:
With `using`  the dispose can be called when runtime wants to. I switched it to explicit Dispose so I know _exactly_ when this happens (so I can also possibly inject traces)

Reply:
That's not true. Dispose will be called at the end of the scope of the method. If it behaved as you say, there wouldn't be any difference between `using` and just throwing an instance away and letting finalizer handle it.

Reply:
Also, you should be able to call explicit Dispose even when you have a using; the using will just call Dispose again eventually, which should be a no-op if Dispose was already called.

Reply:
While I like explicit calls for tests, this is not relevant for the PR so I put it back as it was.